### PR TITLE
Add type hints

### PR DESCRIPTION
**Pull request by dylanjcastillo**
Hi @EmilStenstrom, this is the PR for #390. Let me know what you think!

### CODE REVIEW

File: django_components/types.py [hunk 1]
```diff
@@ -0,0 +1,9 @@
+try:
+    from typing import Annotated
+except ImportError:
+    from typing_extensions import Annotated
```
Comment (EmilStenstrom) [inline django_components/types.py:4]: This needs to be added as a dependency to avoid crashes I think?

File: django_components/types.py [hunk 2]
```diff
@@ -0,0 +1,9 @@
...
+
+html = Annotated[str, "html"]
```
Comment (EmilStenstrom) [inline django_components/types.py:6]: This seems not to be in use?

**Review by EmilStenstrom — COMMENTED** [review]
Looks straightforward! Added two comments, but almost ready to merge!

File: django_components/types.py (hunk 1 above)
Comment (dylanjcastillo) [reply django_components/types.py]: I removed typing-extensions and instead created a simple Annotated class for the cases of Python <3.9. I checked and it still works well to provide the syntax highlighting.

I think it's better not to add additional dependencies if not really necessary, especially for this small use cases.

**Comment by EmilStenstrom** [comment]
Looks great, thanks! :)

**Pull request merged**